src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -502,3 +504,59 @@ export const updateForm = (
     return transformMongoError(error)
   })
 }
+
+export const updateFormSettings = (
+  originalForm: IPopulatedForm,
+  body: SettingsUpdateBody,
+): ResultAsync<
+  Partial<Pick<IFormDocument, keyof SettingsUpdateBody>>,
+  | DatabaseError
+  | DatabaseValidationError
+  | DatabaseConflictError
+  | DatabasePayloadSizeError
+> => {
+  const updateKeys = Object.keys(body) as (keyof SettingsUpdateBody)[]
```

Comment:
can we resolve types at the edge of the application (the controller) instead of doing this sort of thing?

Reply:
This is not really casting right, it's guaranteed to be type (keyof SettingsUpdateBody), but just telling TypeScript as TypeScript is inferencing it as `string[]` instead.